action/Scanning.action

```diff
@@ -0,0 +1,33 @@
+# Goal definition of a single scanning action.
+
+# Pre-defined constants to code the type of scanning action.
+int8 UNKNOWN_TYPE=0  # default value does not code a valid command
+int8 SHELF=1  # detect and localize shelf floors
+int8 BAR_CODE=2  # detect, recognize, and localize bar codes
+int8 SEPARATOR=3  # detect and localize separators
+int8 OBJECT_RECOGNITION=4  # detect, recognize, and localize front object in facing
+int8 OBJECT_COUNTING=5  # count all objects within one facing
+
+# Requested type of scanning action, fill with one of the above constants.
+int8 type
+
+# List of locations that shall be scanned.
+refills_msgs/Location[] locations
+
+---
+# Result definition of a single scanning action.
+
+# Pre-defined constants to code possible runtime failures.
+int8 SUCCESS=-1  # to report successful execution
+int8 UNKNOWN_ERROR=0 # default value codes an unknown error
+
+# Encountered runtime errors.
+int8 error_code
+
+# String field to report human-readable description of runtime failures; optional.
+string error_string
+
+---
+# Feedback definition of a single scanning action.
+refills_msgs/Location current_loc  # location currently being scanned
```

Comment:
the problem with using location.msg in feedback is that it only contains rack OR shelf OR floor etc. but we probably want to know current rack AND shelf AND floor etc right?

Reply:
I do not think so. I imagine this to be a KnowRob ID. If the ID of a shelf floor is returned, then the recipient can query KnowRob to figure out to which shelf or rack this shelf floor belongs, right?